pkg/controller/plan/kubevirt.go

```diff
@@ -974,6 +936,68 @@ func (r *KubeVirt) DeleteHookJobs(vm *plan.VMStatus) (err error) {
 	return
 }
 
+// Get the OpenstackVolumePopulator CustomResource based on the image name.
+func (r *KubeVirt) getOpenstackPopulatorCr(name string) (populatorCr v1beta1.OpenstackVolumePopulator, err error) {
+	populatorCr = v1beta1.OpenstackVolumePopulator{}
+	err = r.Destination.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Plan.Spec.TargetNamespace, Name: name}, &populatorCr)
+	return
+}
+
+// Get the OvirtVolumePopulator CustomResource based on the PVC name.
+func (r *KubeVirt) getOvirtPopulatorCr(name string) (populatorCr v1beta1.OvirtVolumePopulator, err error) {
+	populatorCr = v1beta1.OvirtVolumePopulator{}
+	err = r.Destination.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Plan.Spec.TargetNamespace, Name: name}, &populatorCr)
+	return
+}
+
+// Set the Populator Pod Ownership.
+func (r *KubeVirt) SetPopulatorPodOwnership(vm *plan.VMStatus) (err error) {
+	pvcs, err := r.getPVCs(vm)
+	if err != nil {
+		return
+	}
+	pods, err := r.getPopulatorPods()
+	if err != nil {
+		return
+	}
+	for _, pod := range pods {
+		pvcId := strings.Split(pod.Name, "populate-")[1]
+		for _, pvc := range pvcs {
+			if string(pvc.UID) == pvcId {
+				err = k8sutil.SetOwnerReference(&pvc, &pod, r.Scheme())
+				if err != nil {
+					continue
+				}
+				err = r.Destination.Client.Update(context.TODO(), &pod)
```

Comment:
same question about patch vs update
and we can 'break' here

Reply:
i think it's best effort and if there are any more PVCs to update why not to try?